assets/js/blocks/reviews-by-product/utils.js

```diff
@@ -1,51 +1,71 @@
 /**
  * External dependencies
  */
+import { __, sprintf } from '@wordpress/i18n';
+import { format } from '@wordpress/date';
 import classNames from 'classnames';
 
 /**
- * Render the Reviews by Product block placeholder
+ * Render a review for the Reviews by Product block
  *
  * @param {Object} attributes Block attributes
+ * @param {Object} review Object containing review data.
  *
  * @return {Object} React JSx nodes of the block
  */
-export function renderReviewsByProductPlaceholder( attributes ) {
-	const { className, showProductRating, showReviewerName, showReviewerPicture, showReviewDate } = attributes;
-	const classes = classNames( 'wc-block-reviews-by-product', 'is-loading', className, {
-		'has-picture': showReviewerPicture,
-		'has-name': showReviewerName,
-		'has-rating': showProductRating,
-		'has-date': showReviewDate,
+export function renderReview( attributes, review = {} ) {
+	const { showProductRating, showReviewerName, showReviewerPicture, showReviewDate } = attributes;
+	const { id = null, date_created: dateCreated, rating, review: text = '', reviewer = '', reviewer_avatar_urls: avatarUrls = {} } = review;
+	const isLoading = ! Object.keys( review ).length > 0;
+
+	const classes = classNames( 'wc-block-reviews-by-product__item', {
+		'is-loading': isLoading,
 	} );
+	const starStyle = {
+		width: ( rating / 5 * 100 ) + '%',
+	};
 
 	return (
-		<div className={ classes } aria-hidden>
-			<ul className="wc-block-reviews-by-product__list">
-				<li className="wc-block-reviews-by-product__item">
-					<span className="wc-block-reviews-by-product__text"></span>
-					<div className="wc-block-reviews-by-product__info">
-						{ showReviewerPicture && (
-							<img alt="" src="" srcSet="" className="avatar avatar-60 photo" width="60" height="60" />
-						) }
+		<li className={ classes } key={ id } aria-hidden={ isLoading }>
+			<span
+				className="wc-block-reviews-by-product__text"
+				dangerouslySetInnerHTML={ {
+					// `text` is the `review` parameter returned by the `reviews` endpoint.
+					// It's filtered with `wp_filter_post_kses()`, which removes dangerous HTML tags,
+					// so using it inside `dangerouslySetInnerHTML` is safe.
+					__html: text || '',
```

Comment:
I would like a second opinion about this, to make sure we are not doing anything dangerous here. :point_up:

Reply:
Seems ok. Admin would be able to insert any tag but that is allowed anyway even in PHP comment templates. There isn't really any other way around this.